src/CMDParse.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <map>
+
+#include <tclap/CmdLine.h>
+#include <tclap/Constraint.h>
+
+#include "Utilities.h"
+#include "ModelTypes.h"
+
+#ifndef VERSION
+#error Macro VERSION must be defined!
+#endif
+
+
+namespace EERAModel
+{
+    struct Arguments
+    {
+        bool isLocal = false;
+        std::string local_location=std::string(ROOT_DIR);
+        ModelStructureId stucture;
+        ModelModeId mode = ModelModeId::INFERENCE;
+        std::string output_dir = std::string(ROOT_DIR)+"/outputs";
+    };
+
+    /**
+     * @brief Class for parsing model command line arguments
+     * 
+     * This class uses the TCLAP library to process command line arguments
+     * relating to model structure, execution mode and input locations
+     * 
+     * @param argc number of command line arguments
+     * @param argv argument char** strings
+     */
+    class ArgumentParser
+    {
+        private:
```

Comment:
For consistency with the other code, please put the public members first, and the private after.